### PR TITLE
Implements Bitcoin RPC

### LINKED ISSUE #1245: [RPC] Bitcoin RPC
## list UTXO

1. List all UTXOs by Bitcoin's address. resolve the Bitcoin address to the rooch address and query by `rooch_queryGlobalStates`.
2. Convert the UTXO rooch address to the Bitcoin address(Need the reverse address mapping).
3. Convert the UTXO txid to Bitcoin reverse hex format.
4. We need an on-chain config to record which Bitcoin network we support because the address is different between Bitcoin networks.

## list Inscription

1. Same as above.
2. #1241 

### Reference

#1225

### PR DESCRIPTION
**Pull request by baichuan3**
## Summary

1. List all UTXOs by Bitcoin's address. resolve the Bitcoin address to the rooch address and query by `btc_queryUTXOs`.
2. List all Inscriptions by Bitcoin's address. resolve the Bitcoin address to the rooch address and query by `btc_queryInscriptions`.
3. Convert the rooch address to the Bitcoin address by reverse address mapping
4. Convert the txid to Bitcoin reverse hex format

- Closes #1245

### TODO
1. On-chain config to record which Bitcoin network we support because the address is different between Bitcoin networks

### CODE REVIEW

**Comment by baichuan3** [comment]
The BitcoinAddress binary resolve from Move does not match the address displayed in Rust.

The reproduction case is as follows:

1. Get any BitcoinAdress bytes address from Move contract light_client
https://github.com/rooch-network/rooch/blob/2be5d2d8352ca2aaa724cde91b034e54886ce8a1/frameworks/bitcoin-move/sources/light_client.move#L191-L198

```
  0x3::bitcoin_address::BitcoinAddress {
       bytes: 0x012045966003624094dae2deeb30815eedd38f96c45c3fdb1261f5d697fc4137e0de
     }
```

The resolved rooch address relative above Bitcoin Address is 0xf337f0d7f03324fa19874a9a0b41e0332616797497b8e9061746444e53fe433c

Among them, the binary corresponding to BitcoinAddress is

[1, 32, 233, 38, 245, 228, 103, 162, 211, 102, 231, 86, 199, 140, 8, 24, 164, 37, 190, 174, 60, 18, 29, 197, 251 , 213, 82, 220, 198, 158, 116, 108, 39, 6]

The first byte is the version opcode, the second byte is the push opcode, and the following bytes are the witness program. The problem now is that there is a judgment in Payload::from_scripts() that is_witness_program() cannot pass.

```
     pub fn is_witness_program(&self) -> bool {
         // A scriptPubKey (or redeemScript as defined in BIP16/P2SH) that consists of a 1-byte
         // push opcode (for 0 to 16) followed by a data push between 2 and 40 bytes gets a new
         // special meaning. The value of the first push is called the "version byte". The following
         // byte vector pushed is called the "witness program".
         let script_len = self.0.len();
         if !(4..=42).contains(&script_len) {
             return false;
         }
         let ver_opcode = Opcode::from(self.0[0]); // Version 0 or PUSHNUM_1-PUSHNUM_16
         let push_opbyte = self.0[1]; // Second byte push opcode 2-40 bytes
         WitnessVersion::try_from(ver_opcode).is_ok()
             && push_opbyte >= OP_PUSHBYTES_2.to_u8()
             && push_opbyte <= OP_PUSHBYTES_40.to_u8()
             // Check that the rest of the script has the correct size
             && script_len - 2 == push_opbyte as usize
     }
```

This step fails with `WitnessVersion::try_from(ver_opcode).is_ok()`.
Since version opcode = 1, the following conditional judgment will fail.


```
impl TryFrom<Opcode> for WitnessVersion {
     type Error = TryFromError;

     fn try_from(opcode: Opcode) -> Result<Self, Self::Error> {
         match opcode.to_u8() {
             0 => Ok(WitnessVersion::V0),
             version if version >= OP_PUSHNUM_1.to_u8() && version <= OP_PUSHNUM_16.to_u8() =>
                 WitnessVersion::try_from(version - OP_PUSHNUM_1.to_u8() + 1),
             invalid => Err(TryFromError { invalid }),
         }
     }
}
```

2. Rust test cases locate https://github.com/rooch-network/rooch/blob/25866e43b48c2acb9a157d8e6813e8b5b028f1c4/crates/rooch-types/src/address.rs#L697-L772
3. Rooch server command `rooch server start --btc-rpc-url http://127.0.0.1:18443 --btc-rpc-username roochuser --btc-rpc-password roochpass --btc-start-block-height 0`
4. rust test command `cd rooch/crates/rooch-types; cargo test --tests -- --nocapture`

**Comment by baichuan3** [comment]
‘’'
curl --location 'http://localhost:50051' \
--header 'Content-Type: application/json' \
--data '{
    "id": 101,
    "jsonrpc": "2.0",
    "method": "rooch_queryGlobalStates",
    "params": [
        {"object_type_with_owner":{"object_type":"0x4::utxo::UTXO","owner":"0xf8320c07f9e30ea1d4f9e4d7abf07ced414e2f67cdcac03dff86a31f9ee03a41"}}, null, "3", true
    ]
}'
'''

```
{
    "jsonrpc": "2.0",
    "result": {
        "data": [
            {
                "object_id": "0xa66da28355ce244b6146fbf45fd24f003fffac9ccce00beb1b59f5e78373870e",
                "owner": "0xf8320c07f9e30ea1d4f9e4d7abf07ced414e2f67cdcac03dff86a31f9ee03a41",
                "flag": 0,
                "value": {
                    "abilities": 8,
                    "type": "0x4::utxo::UTXO",
                    "value": {
                        "seals": {
                            "abilities": 7,
                            "type": "0x2::simple_multimap::SimpleMultiMap<0x1::string::String, 0x2::object::ObjectID>",
                            "value": {
                                "data": []
                            }
                        },
                        "txid": "0x2d5489bdbb7c4f39cf0395a35d914445fa58f616cdce868a2d8915e37310c3b0",
                        "value": "2500000295",
                        "vout": 0
                    }
                },
                "object_type": "0x4::utxo::UTXO",
                "key_type": null,
                "size": 0,
                "tx_order": 407,
                "state_index": 6,
                "created_at": 0,
                "updated_at": 0
            },
            {
                "object_id": "0xb89184fa53013062525957e26ba92ae6f3405965d4f70a04d6ddf9a8e3193014",
                "owner": "0xf8320c07f9e30ea1d4f9e4d7abf07ced414e2f67cdcac03dff86a31f9ee03a41",
                "flag": 0,
                "value": {
                    "abilities": 8,
                    "type": "0x4::utxo::UTXO",
                    "value": {
                        "seals": {
                            "abilities": 7,
                            "type": "0x2::simple_multimap::SimpleMultiMap<0x1::string::String, 0x2::object::ObjectID>",
                            "value": {
                                "data": []
                            }
                        },
                        "txid": "0xda9f977c7271521b453d2ea9f37ab92279605acbdfb17dd007e2d2794de56a9e",
                        "value": "2500000000",
                        "vout": 0
                    }
                },
                "object_type": "0x4::utxo::UTXO",
                "key_type": null,
                "size": 0,
                "tx_order": 405,
                "state_index": 3,
                "created_at": 0,
                "updated_at": 0
            },
            {
                "object_id": "0x725d11d43d5026a179ad140f7e81d73c2e8a0b1071c41b8097eefd830c4ca932",
                "owner": "0xf8320c07f9e30ea1d4f9e4d7abf07ced414e2f67cdcac03dff86a31f9ee03a41",
                "flag": 0,
                "value": {
                    "abilities": 8,
                    "type": "0x4::utxo::UTXO",
                    "value": {
                        "seals": {
                            "abilities": 7,
                            "type": "0x2::simple_multimap::SimpleMultiMap<0x1::string::String, 0x2::object::ObjectID>",
                            "value": {
                                "data": []
                            }
                        },
                        "txid": "0xbc12cb9b31be6b4ccb6a4877ad69f84a70119b08b62a09b1c318b16737313022",
                        "value": "2500000000",
                        "vout": 0
                    }
                },
                "object_type": "0x4::utxo::UTXO",
                "key_type": null,
                "size": 0,
                "tx_order": 403,
                "state_index": 0,
                "created_at": 0,
                "updated_at": 0
            }
        ],
        "next_cursor": {
            "tx_order": 403,
            "state_index": 0
        },
        "has_next_page": true
    },
    "id": 101
}
```

```
curl --location 'http://localhost:50051' \
--header 'Content-Type: application/json' \
--data '{
    "id": 101,
    "jsonrpc": "2.0",
    "method": "rooch_queryGlobalStates",
    "params": [
        {"multi_chain_address":{"multichain_id":0, "address":"bc1pgktxqqmzgz2d4ck7avcgzhhd6w8ed3zu8ld3yc0466tlcsfhur0qj3y0wm"}}, null, "10", true
    ]
}'
```

```
{
    "jsonrpc": "2.0",
    "result": {
        "data": [
            {
                "object_id": "0xa66da28355ce244b6146fbf45fd24f003fffac9ccce00beb1b59f5e78373870e",
                "owner": "0xf8320c07f9e30ea1d4f9e4d7abf07ced414e2f67cdcac03dff86a31f9ee03a41",
                "flag": 0,
                "value": {
                    "abilities": 8,
                    "type": "0x4::utxo::UTXO",
                    "value": {
                        "seals": {
                            "abilities": 7,
                            "type": "0x2::simple_multimap::SimpleMultiMap<0x1::string::String, 0x2::object::ObjectID>",
                            "value": {
                                "data": []
                            }
                        },
                        "txid": "0x2d5489bdbb7c4f39cf0395a35d914445fa58f616cdce868a2d8915e37310c3b0",
                        "value": "2500000295",
                        "vout": 0
                    }
                },
                "object_type": "0x4::utxo::UTXO",
                "key_type": null,
                "size": 0,
                "tx_order": 407,
                "state_index": 6,
                "created_at": 0,
                "updated_at": 0
            },
            {
                "object_id": "0xb89184fa53013062525957e26ba92ae6f3405965d4f70a04d6ddf9a8e3193014",
                "owner": "0xf8320c07f9e30ea1d4f9e4d7abf07ced414e2f67cdcac03dff86a31f9ee03a41",
                "flag": 0,
                "value": {
                    "abilities": 8,
                    "type": "0x4::utxo::UTXO",
                    "value": {
                        "seals": {
                            "abilities": 7,
                            "type": "0x2::simple_multimap::SimpleMultiMap<0x1::string::String, 0x2::object::ObjectID>",
                            "value": {
                                "data": []
                            }
                        },
                        "txid": "0xda9f977c7271521b453d2ea9f37ab92279605acbdfb17dd007e2d2794de56a9e",
                        "value": "2500000000",
                        "vout": 0
                    }
                },
                "object_type": "0x4::utxo::UTXO",
                "key_type": null,
                "size": 0,
                "tx_order": 405,
                "state_index": 3,
                "created_at": 0,
                "updated_at": 0
            },
            {
                "object_id": "0x725d11d43d5026a179ad140f7e81d73c2e8a0b1071c41b8097eefd830c4ca932",
                "owner": "0xf8320c07f9e30ea1d4f9e4d7abf07ced414e2f67cdcac03dff86a31f9ee03a41",
                "flag": 0,
                "value": {
                    "abilities": 8,
                    "type": "0x4::utxo::UTXO",
                    "value": {
                        "seals": {
                            "abilities": 7,
                            "type": "0x2::simple_multimap::SimpleMultiMap<0x1::string::String, 0x2::object::ObjectID>",
                            "value": {
                                "data": []
                            }
                        },
                        "txid": "0xbc12cb9b31be6b4ccb6a4877ad69f84a70119b08b62a09b1c318b16737313022",
                        "value": "2500000000",
                        "vout": 0
                    }
                },
                "object_type": "0x4::utxo::UTXO",
                "key_type": null,
                "size": 0,
                "tx_order": 403,
                "state_index": 0,
                "created_at": 0,
                "updated_at": 0
            }
        ],
        "next_cursor": {
            "tx_order": 403,
            "state_index": 0
        },
        "has_next_page": true
    },
    "id": 101
}
```

File: moveos/moveos-types/src/moveos_std/simple_multimap.rs [hunk 1]
```diff
@@ -14,8 +14,6 @@ use move_core_types::{
 };
 use serde::{Deserialize, Serialize};
 
-pub const MODULE_NAME: &IdentStr = ident_str!("simple_multimap");
```
Comment (jolestar) [inline moveos/moveos-types/src/moveos_std/simple_multimap.rs:17]: Why remove this constant?

**Review by jolestar — APPROVED** [review]

File: moveos/moveos-types/src/moveos_std/simple_multimap.rs (hunk 1 above)
Comment (baichuan3) [reply moveos/moveos-types/src/moveos_std/simple_multimap.rs]: fixup

**Pull request merged**